source/base/utilities.cc

```diff
@@ -466,20 +466,13 @@ namespace Utilities
   unsigned int
   needed_digits(const unsigned int max_number)
   {
-    if (max_number < 10)
+    if (max_number > 0)
+      return static_cast<int>(
+        std::ceil(std::log10(std::fabs(max_number + 0.1))));
+    else
       return 1;
```

Comment:
We can get rid of this `else` clause. I think clang-tidy will complain about it so this will placate one of the CI builds.

Reply:
done in https://github.com/dealii/dealii/pull/8924/commits/20f131f20cfe477e162501d34fece92307858667